Blog-projects/Widget-window-in-Qt-Quick-app/fontcontrolswidgetsform.cpp

```diff
@@ -0,0 +1,48 @@
+#include "fontcontrolswidgetsform.h"
+#include "ui_fontcontrolswidgetsform.h"
+
+FontControlsWidgetsForm::FontControlsWidgetsForm(QWidget *parent)
+    : QWidget(parent)
+    , ui(new Ui::FontControlsWidgetsForm)
+{
+    ui->setupUi(this);
+    connect(ui->lineEdit, &QLineEdit::textEdited, this, &FontControlsWidgetsForm::lineEditTextChanged);
+    connect(ui->fontComboBox, &QFontComboBox::currentFontChanged, this, &FontControlsWidgetsForm::currentFontChanged);
+    connect(ui->pushButton, &QPushButton::clicked, this, &FontControlsWidgetsForm::buttonClicked);
+}
+
+FontControlsWidgetsForm::~FontControlsWidgetsForm()
+{
+    delete ui;
+}
+
+const QString FontControlsWidgetsForm::text()
+{
+    return ui->lineEdit->text();
+}
+
+void FontControlsWidgetsForm::lineEditTextChanged(const QString &arg1)
+{
+    emit textChanged();
+}
+
+void FontControlsWidgetsForm::setText(const QString &text)
+{
+    ui->lineEdit->setText(text);
+    emit textChanged();
```

Comment:
Actually, if you were connecting to textChanged instead of textEdited, you could remove this emit, it would be done automatically by setText -> textChanged -> lineEditTextChanged()  (obviously, don't rename the slot if you change the signal you connect to)

Reply:
Nice catch!